xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/ListClass.java

```diff
@@ -649,16 +649,19 @@ private String getDisplayValue(String value, String name, Map<String, ListItem>
         }
         String msgname = getFieldFullName() + "_" + value;
         String newresult = localizePlain(msgname);
+        String syntax = context.getDoc().getSyntax().toIdString();
         if (newresult == null) {
             msgname = "option_" + name + "_" + value;
             newresult = localizePlain(msgname);
             if (newresult == null) {
                 msgname = "option_" + value;
                 newresult = localizePlain(msgname);
                 if (newresult == null) {
-                    newresult = displayValue;
+                    newresult = context.getDoc().getRenderedInlineContent(displayValue, syntax);
```

Comment:
Not sure to understand why it needed a change there.

Reply:
I don't understand what you are doing here, a list does not contain wiki content.

Reply:
I thought it could. Is it safe to assume that the content of ListClass and StringClass is always using plain/1.0?

Reply:
I have my answer in your next comment. I'll adapt accordingly.